### DISCUSSION TITLE
2 or 3 output for one port

### DISCUSSION THREAD

**Discussion by WiilyLin** [nasa/fprime#2174] [Q&A] [answered]
When I want to connect the components, I found that output for one port can be only one. It cannot 2 or 3 output for one port. But input for one port can lager than one. Is that right? Following is result in topology,
for 2 or 3 output for one port. and one input for port.

>       Project.Manager_aocs.AOCStoDriver -> Project.AOCS_Magnetometer.AOCSinput
>       Project.Manager_aocs.AOCStoDriver -> Project.AOCS_Accelerometer.AOCSinput

and it false. 

>       command_test.command_port -> accelerometer.commandin
>       task.command_port -> accelerometer.commandin

And it can work. 
So I want to ask how to make the upper one work? Or maybe the only way is to make a lot of port?

**Comment by Joshua-Anderson** [comment, marked as answer]
F' output ports can be arrays ([example](https://github.com/nasa/fprime/blob/devel/Svc/CmdDispatcher/CmdDispatcher.fpp#L21)).

Instead of writing to a single output port, you can make the output port an array and loop over and write to all the output ports in the array.

**Reply by dav-oss** [reply]
Not disagreeing with you but a vector(s) might be more suitable than normal array(s).
